### PR TITLE
Nix fix

### PR DESCRIPTION
**Pull request by Reasonable-Solutions**
@x10an14-nav

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -35,7 +35,10 @@
 
       # Target musl when building on 64-bit linux
       buildTarget =
-        {"x86_64-linux" = "x86_64-unknown-linux-musl";}.${system}
+        {
+          "x86_64-linux" = "x86_64-unknown-linux-musl";
+        }
+        .${system}
```
Comment (x10an14-nav) [inline flake.nix:41]: Eh, don't like, but can live with =P

File: flake.nix [hunk 2]
```diff
@@ -53,19 +56,16 @@
...
-              # uncomment to debug:
-              # builtins.trace "${relPath}: ${lib.boolToString includePath}"
-              includePath
-          );
+          filter = path: type: let
+            relPath = pkgs.lib.removePrefix basePath (toString path);
+            includePath =
+              (type == "directory")
+              || pkgs.lib.any (re: builtins.match re relPath != null)
+              regexes;
```
Comment (x10an14-nav) [inline flake.nix:64]: Very much lkike =D

File: flake.nix [hunk 3]
```diff
@@ -53,19 +56,16 @@
...
+            # uncomment to debug:
+            # builtins.trace "${relPath}: ${lib.boolToString includePath}"
```
Comment (x10an14-nav) [inline flake.nix:66]: Moving this into the `let ... in` statement is just plain wrong, no? o.0

File: flake.nix [hunk 4]
```diff
@@ -85,7 +85,9 @@
         });
 
       # Compile and cache only workspace code (seperately from 3rc party dependencies)
-      package-file-filter = dep-files-filter ++ ["src/.+\.rs" "src/test_data/.+\.yaml"];
+      package-file-filter =
```
Comment (x10an14-nav) [inline flake.nix:88]: Personally not a fan of dangling ` =\n`, but I guess I can live with it.

File: flake.nix [hunk 5]
```diff
@@ -125,7 +127,10 @@
             rustToolchain
             # cargo-package # Comment in when you want tests to run on every new shell
           ]
-          ++ pkgs.lib.optionals (pkgs.stdenv.isLinux) (with pkgs; [cargo-watch]); # Currently broken on macOS
+          ++ pkgs.lib.optionals pkgs.stdenv.isDarwin
+          [pkgs.darwin.apple_sdk.frameworks.Security]
+          ++ pkgs.lib.optionals pkgs.stdenv.isLinux
+          (with pkgs; [cargo-watch]); # Currently broken on macOS
```
Comment (x10an14-nav) [inline flake.nix:133]: Would you happen to have a chance to test if this still holds true?
Ref: https://nav-it.slack.com/archives/C8H9NLWG0/p1686335751598759

File: flake.nix [hunk 6]
```diff
@@ -125,7 +127,10 @@
             rustToolchain
             # cargo-package # Comment in when you want tests to run on every new shell
           ]
-          ++ pkgs.lib.optionals (pkgs.stdenv.isLinux) (with pkgs; [cargo-watch]); # Currently broken on macOS
+          ++ pkgs.lib.optionals pkgs.stdenv.isDarwin
```
Comment (x10an14-nav) [inline flake.nix:130]: Ooooh, okay! Any github nixpkgs/issue you could link to? I imagine that's where you found this workaround?
Curious as to why this is necessary =)

**Review by x10an14-nav — COMMENTED** [review]
LGTM, commented just to elicit some dialogue, curious about perspectives here.

Not approving just so that it doesn't close (yet).
(I was told the changes already got pushed to main)

File: flake.nix (hunk 3 above)
Comment (x10an14-nav) [reply flake.nix:66]: It seems that if you manually move it back and re-run `nixfmt`, they (both `alejandra` and `nixfmt`) permit/keep it.

**Comment by x10an14-nav** [comment]
Strange, they look identical to me. I don't mind switching =)
```
-> $ git diff alejandra-nix-format..nixfmt-nix-format

───────────────┐
flake.nix:153: │
───────────────┘
 153⋮ 153│      packages.default = cargo-package;
 154⋮ 154│
 155⋮ 155│      # Now `nix fmt` works!
 156⋮    │      formatter = pkgs.alejandra;
    ⋮ 156│      formatter = pkgs.nixfmt;
 157⋮ 157│    });
 158⋮ 158│}
[2023-06-13 16:42:31] 0 x10an14@home-desktop:~/Documents/nav/github/navikt/mutilator
-> $ git log --oneline ^main alejandra-nix-format nixfmt-nix-format
e530b4c (HEAD -> nixfmt-nix-format) refactor: nixfmt nixfmt'ed
0b618c8 (alejandra-nix-format) refactor: Alejandra nix format nixfmter.
```

File: flake.nix (hunk 5 above)
Comment (Reasonable-Solutions) [reply flake.nix:133]: sure, I will have a look

File: flake.nix (hunk 6 above)
Comment (Reasonable-Solutions) [reply flake.nix:130]: I am sure I can find one, if nothing else ;p

**Comment by Reasonable-Solutions** [comment]
> Strange, they look identical to me. I don't mind switching =)
> 
> ```
> -> $ git diff alejandra-nix-format..nixfmt-nix-format
> 
> ───────────────┐
> flake.nix:153: │
> ───────────────┘
>  153⋮ 153│      packages.default = cargo-package;
>  154⋮ 154│
>  155⋮ 155│      # Now `nix fmt` works!
>  156⋮    │      formatter = pkgs.alejandra;
>     ⋮ 156│      formatter = pkgs.nixfmt;
>  157⋮ 157│    });
>  158⋮ 158│}
> [2023-06-13 16:42:31] 0 x10an14@home-desktop:~/Documents/nav/github/navikt/mutilator
> -> $ git log --oneline ^main alejandra-nix-format nixfmt-nix-format
> e530b4c (HEAD -> nixfmt-nix-format) refactor: nixfmt nixfmt'ed
> 0b618c8 (alejandra-nix-format) refactor: Alejandra nix format nixfmter.
> ```

He, that is funny. I use whatever formatter aphelia-mode defaults to. Let me do an investigate on this.

**Comment by x10an14-nav** [comment]
Waiting on me here? Not sure what else is left? Should we just close?

**Comment by mortenlj** [comment]
If the changes here are already pushed to main, I suggest just closing this unless there are any unresolved points to discuss?

Any other discussion can continue on Slack IMHO.

**Pull request closed without merging**